### ISSUE TITLE
How to change my profile picture?

### ISSUE THREAD

**Issue by baldantas** [tgalal/yowsup#620]
I'm new to the forum,
I have configured the Yowsup, and all the examples of DEMO could test. Is there any example or instruction to specify the profile photo?

God bless you all.

**Comment by makinzoid** [comment]
Hi! Got the same question. Thanks in advance!

**Comment by tgalal** [comment]
Sorry this is still work in progress

**Comment by aesedepece** [comment]
:+1: to this feature request.

@tgalal Maybe we could make a list of dropped features from yowsup1 to yowsup2 so that people eager to contribute can know where to put their efforts.

**Comment by jlguardi** [comment]
I'm working on it but it seems that profile picture setting is not made with this message:
```
DEBUG:yowsup.layers.logger.layer:tx:
<iq xmlns="w:profile:picture" type="set" id="1">
<picture type="image" id="2">
����JFIF��
....
HEX:...
</picture>
</iq>

[connected]:DEBUG:yowsup.layers.logger.layer:rx:
<iq type="error" from="3xxxxx@s.whatsapp.net" id="1">
<error text="not-acceptable" code="406">
</error>
</iq>
```

I've also tryed with 'to' attribute but same error.

Can someone make a tutorial about howto sniff original whatsapp? I can get stream without https encryption but I don't know how to decode it. Is it easy?

**Comment by aguiraf** [comment]
@jlguardi could be an image format issue?
With yowsup1 I succeded change the profile picture only with a 480x480px jpeg image...

**Comment by jlguardi** [comment]
I've tried with an image JPEG 480x480 480x480+0+0 8-bit DirectClass 17.3KB 0.000u 0:00.000 and also an image downloaded directly from other WA contact.

Not working yet! Still "not-acceptable" response.

**Comment by aesedepece** [comment]
I already got it working and will push a PR soon :wink:

**Comment by aguiraf** [comment]
+1￼

**Comment by RiccoRammer** [comment]
Any news on this feature?

**Comment by aesedepece** [comment]
@RiccoRammer  Oops. Totally forgot about it. I'll push the PR asap.

**Comment by aesedepece** [comment]
Wait. This was solved by @tgalal in f9b88e51f764f16e50d506938b591c5522f69d49, right?

**Comment by landantas** [comment]
Hey guys, i'm with this problem too.
I make this changes of the fix, but now, adding this changes, we don't receive data at "contact_picture" function, function that get the picture from contact, and not succeed change the profile picture yet.

```
def setPicture(self, path):
        if self.connected:
            print "Setting picture: ", path

            f = open(path, 'rb')
            imageData = f.read()
            imageData = bytearray(imageData)
            f.close()

            entity = PictureIqProtocolEntity(jid = self.aliasToJid(self.phone), _id = ("set_pic-"+str(int(time.time()))), type = "set")
            entity.setPictureData(imageData)
            print entity
            self.toUpper(entity)
        else:
            print "Not connected yet"
```

It works without the changes, but not after the fix:

```
@ProtocolEntityCallback("iq")
    def onIq(self, entity):
        if hasattr(entity, "pictureData"):
            UU = entity.getPictureData().encode("base64")

            envio = entity.getFrom().split('@')

            self.socketIO.emit("profilePicture", envio[0], entity.getPictureId(), UU)
        else:
            print "IQ:\n", entity

    def contact_picture(self, jid):
        if self.connected:
            #print "Getting picture from: ", self.aliasToJid(jid)
            entity = PictureIqProtocolEntity(self.aliasToJid(jid))
            self.toLower(entity)
```

Stay with God!
